roachtest: sqlsmith/setup=empty/setting=no-mutations failed

roachtest.sqlsmith/setup=empty/setting=no-mutations [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3248647&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3248647&tab=artifacts#/sqlsmith/setup=empty/setting=no-mutations) on master @ [31af9e32a55a166166e9ba9c5327b7cd847ae236](https://github.com/cockroachdb/cockroach/commits/31af9e32a55a166166e9ba9c5327b7cd847ae236):


```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/sqlsmith/setup=empty/setting=no-mutations/run_1
	sqlsmith.go:217,sqlsmith.go:252,test_runner.go:777: error: pq: internal error: no volatility for cast tuple{timestamptz, int[], unknown}::tuple{timestamptz, int2[], unknown}
		stmt:
		WITH
			with_166382 (col_927225)
				AS (
					SELECT
						*
					FROM
						(VALUES (('0001-01-01 00:00:00+00:00':::TIMESTAMPTZ, ARRAY[]:::INT2[], NULL)))
							AS tab_537300 (col_927225)
					EXCEPT ALL
						SELECT
							*
						FROM
							(
								VALUES
									(
										('294276-12-31 23:59:59.999999+00:00':::TIMESTAMPTZ, ARRAY[(-32768):::INT8,0:::INT8,25203:::INT8], NULL)
									)
							)
								AS tab_537301 (col_927226)
				)
		SELECT
			tab_537302.col_927227 AS col_927228
		FROM
			(
				VALUES
					((-0.2372959852218628):::FLOAT8),
					(0.3741636276245117:::FLOAT8),
					((-0.2869168221950531):::FLOAT8),
					(1.1908493041992188:::FLOAT8),
					((-1.7612881660461426):::FLOAT8),
					((-0.4135175943374634):::FLOAT8)
			)
				AS tab_537302 (col_927227)
		WHERE
			true;
```
<details><summary>Reproduce</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/roachtest)

</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*sqlsmith/setup=empty/setting=no-mutations.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/logictest/testdata/logic_test/tuple
+++ b/pkg/sql/logictest/testdata/logic_test/tuple
@@ -968,3 +968,14 @@ SELECT (ARRAY[t58439.*][2]).* FROM t58439
 NULL  NULL
 NULL  NULL
 NULL  NULL
+
+# Regression test for #68308. Do not internally error with a cast of unknown to
+# unknown in a tuple.
+subtest regression_68308
+
+query T
+SELECT (1::INT2, NULL)
+UNION
+SELECT (1::INT, NULL)
+----
+(1,)

--- a/pkg/sql/sem/tree/casts.go
+++ b/pkg/sql/sem/tree/casts.go
@@ -335,6 +335,9 @@ var validCasts = []castInfo{
 	{from: types.IntFamily, to: types.OidFamily, volatility: VolatilityStable, ignoreVolatilityCheck: true},
 	{from: types.OidFamily, to: types.OidFamily, volatility: VolatilityStable},
 
+	// Casts to UnknownFamily.
+	{from: types.UnknownFamily, to: types.UnknownFamily, volatility: VolatilityImmutable},
+
 	// Casts to UuidFamily.
 	{from: types.UnknownFamily, to: types.UuidFamily, volatility: VolatilityImmutable},
 	{from: types.StringFamily, to: types.UuidFamily, volatility: VolatilityImmutable},

